### PR TITLE
Initialize logging when running tests

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

**Review by dain — APPROVED** [review]

File: src/main/java/io/airlift/junit/InitializeLogging.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class InitializeLogging
+        implements Extension
+{
+    static {
+        Logging logging = Logging.initialize();
+
+        // disable logging from testcontainers by default
+        logging.setLevel("tc", Level.OFF);
+        logging.setLevel("org.testcontainers", Level.OFF);
```
Comment (electrum) [inline src/main/java/io/airlift/junit/InitializeLogging.java]: Make this `WARN`. We need to see errors from wait strategy, etc.

File: src/main/java/io/airlift/junit/InitializeLogging.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.jupiter.api.extension.Extension;
+
+public class InitializeLogging
+        implements Extension
+{
+    static {
+        Logging logging = Logging.initialize();
+
+        // disable logging from testcontainers by default
+        logging.setLevel("tc", Level.OFF);
```
Comment (electrum) [inline src/main/java/io/airlift/junit/InitializeLogging.java]: Remove this. It’s useful to see when containers are pulled, starting, how long they took, etc.

**Review by electrum — CHANGES_REQUESTED** [review]
If we’re going to specify logging defaults here, we need a way for users to override them. Maybe add a system property prefix for logger levels.

As a future improvement, it would also be useful to allow tests to specify additional defaults.

File: src/main/java/io/airlift/junit/InitializeLogging.java (hunk 1 above)
Comment (martint) [reply src/main/java/io/airlift/junit/InitializeLogging.java]: Which errors? As far as I can tell, those will propagate all the way up the stack and cause tests to fail with an exception that will get printed by the testing framework.

File: src/main/java/io/airlift/junit/InitializeLogging.java (hunk 2 above)
Comment (martint) [reply src/main/java/io/airlift/junit/InitializeLogging.java]: Sure, I can remove it for now. However, I find the logs from TestContainers  to be very verbose. Combined with everything else, it's the proverbial "death by a thousand cuts" that makes it impossible to find what caused a test failure in the sea of logs.

**Comment by martint** [comment]
I dropped the changes to the logging levels. For now, this just initializes the logging system so we get better formatted log lines. @electrum, please take a look.

**Review by electrum — APPROVED** [review]

**Pull request merged**